core/src/collections.rs

```diff
@@ -26,29 +28,107 @@ impl MongoCollections {
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
     pub fn list_tables(
-        _client: &MongoConnection,
+        mongo_connection: &MongoConnection,
         _query_timeout: Option<i32>,
-        _db_name_filter: &str,
-        _collection_name_filter: &str,
+        db_name_filter: &str,
+        collection_name_filter: &str,
     ) -> Self {
-        unimplemented!()
+        let mut databases: Vec<CollectionsForDb> = vec![];
```

Comment:
lets use `with_capacity` since we know the length of this from the `list_database_names` call. Should save some reallaocs